src/webxdc/maps_integration.rs

```diff
@@ -0,0 +1,279 @@
+//! # Maps Webxdc Integration.
+//!
+//! A Maps Webxdc Integration uses `sendUpdate()` and `setUpdateListener()` as usual,
+//! however, it agrees with the core on the following update format:
+//!
+//! ## Setting POIs via `sendUpdate()`
+//!
+//! ```json
+//! payload: {
+//!     action: "pos",
+//!     lat:    53.550556,
+//!     lng:    9.993333,
+//!     label:  "my poi"
+//! }
+//! ```
+//!
+//! Just sent POI are received via `setUpdateListener()`, as well as old POI.
+//!
+//! ## Receiving Locations via `setUpdateListener()`
+//!
+//! ```json
+//! payload: {
+//!     action:     "pos",
+//!     lat:        47.994828,
+//!     lng:        7.849881,
+//!     timestamp:  1712928222,
+//!     contactId:  123,    // can be used as a unique ID to differ tracks etc
+//!     name:       "Alice",
+//!     color:      "#ff8080",
+//!     independent: false, // false: current or past position of contact, true: a POI
+//!     label:       ""     // used for POI only
+//! }
+//! ```
+//!
+//! For messenger implementors adding support for these Webxdc:
+//! New locations, are announced by `DC_EVENT_LOCATION_CHANGED`
```

Comment:
of course, if we can manage to also emit DC_EVENT_WEBXDC_STATUS_UPDATE so that UI does not need to take care of that, that'd be nice - however, i do not have a good idea how to do that without too much effort.

but that is really a minor, i mean, the whole integration of the map into eg. iOS is only 50 LOC - including adding/removing observers. and some little things in UI are needed anyways - eg. you also want to show a different title and you do not want to have the standard menu
a bit tricky might also be be when to emit DC_EVENT_WEBXDC_STATUS_UPDATE - core does not really know about if the xdc is opened of not. emit it always?

> Especially since this note is hidden somewhere in the implementation docs and not API documentation.

the maps-integration-API is currently only internal anyways, the reference for @adbenitez and @nicodh is  that file. i expect JSON calls to be changed anyways

Reply:
having a closer look: LocationChanged is emitted only 4 times - maybe we can just emit another  DC_EVENT_WEBXDC_STATUS_UPDATE unconditionally?

that'd be a simple solution :)

Reply:
> Yes, just emit status update for this xdc if it is currently "initialized" for this chat or is global.

k, i add a commit

Reply:
done :) this removes another 30 LOC on iOS, integration is really simple now 👌

Reply:
btw,  i think, the "data2 serial" in DC_EVENT_WEBXDC_STATUS_UPDATE seems to be a relict from the first webxdc implementation having races; we can probably just drop that in another PR

Reply:
it is not a relict, it was needed by bots that work with webxdc, so they need to know the ID/serial to load the status update to process it in that event

Reply:
ah, `data2` was removed by https://github.com/deltachat/deltachat-core-rust/pull/3081 for **replication safety** - but then re-added for bots at https://github.com/deltachat/deltachat-core-rust/pull/5461 :/

we can leave it for now, however, we should strongly discourage using data2 as it may also stand in the way of other things. is it used in many bots?

in case of integrations (that are no webxdc bots), it is fine to leave that unset, so, not really an issue of this PR

Reply:
it was deleted and reintroduced at: https://github.com/deltachat/deltachat-core-rust/pull/3215

Reply:
> we can leave it for now, however, we should strongly discourage using data2 as it may also stand in the way of other things. is it used in many bots?

at least all my bots that work with webxdc (ex. public bots, the bot+webxdc example) the webxdc store is a rust bot and maybe there there is more control over the API, but as long as there is no other way for bots to get this information it would be too bad if it get removed, bots need to react per status update, the whole way the webxdc api exist is focused in the webview clients implementations and it is already uncomfortable to work with with bots